utils/utils.go

```diff
@@ -0,0 +1,34 @@
+package utils
+
+import (
+	"context"
+	"database/sql"
+	"medical-clinic/models"
+	"net/http"
+	"strconv"
+)
+
+func IsAdminAllowed(w http.ResponseWriter, r *http.Request, db *sql.DB) bool {
```

Comment:
I recommend using this as a middleware. Here is an example of how using middleware with gorilla mux
https://gopherslab.com/insights/gorilla-mux-middleware-in-go/